mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/core/CoreConfiguration.java

```diff
@@ -78,4 +77,12 @@ public interface CoreConfiguration {
     @Config("mantis.asyncHttpClient.readTimeoutMs")
     @Default("10000")
     int getAsyncHttpClientReadTimeoutMs();
+
+    @Config("mantis.leader.elector.factory")
+    @Default("io.mantisrx.server.core.master.LocalLeaderFactory")
```

Comment:
A few issues I see here:
* This leaderElector is only for control plane so it doesn't need to be in coreConfig (and all the workerConfig related classes) for worker/agents.
* We are actually trying to deprecate/move away from config-injected objects. In this case the if the electorFactory is only used in control-plane startup I assume it can be either a string in masterConfig directly.

Reply:
Happy to remove this part, I noticed it today as well.

I'm not a huge fan of the DI with Skife today (and it's a little out of date). So I'm happy to hear that position.

What are the current thoughts on how we should think about extending Mantis then? I started with the view point I should only take a dependency on core, this way client or server could later bundle my jar. The only option I could find was the object-injection with Skife today. Should I be looking for another pattern here? 

Potentially we could separate each component (@crioux-stripe and I started there with store). That felt pretty heavy weight for each store and we duplicated a lot of DynamoDB config, so this time I went with just an extension for DynamoDB in a gradle project. We could separate each feature into subprojects in the Gradle project and choose different dependencies for each subproject:

```
mantis-ext-dynamodb
  * common (common dynamodb config each subject depends on common)
  * server [subproject only depends on mantis-control-plane-server] (contains store and leader elector) 
  * leader-monitor [subproject depends on mantis-control-plane-core] (contains only monitor)
```

The above feels pretty heavy weight, but perhaps more precise? We'd have to document this somewhere as well and it's sort of straightforward, but definitely not clear when you look at the current source code.

Reply:
> I started with the view point I should only take a dependency on core, this way client or server could later bundle my jar. The only option I could find was the object-injection with Skife today. Should I be looking for another pattern here?
Are you saying you don't take a dependency on mantis-control-plane-server, so you don't have access to the MasterConfiguration?

For skife injection, we don't have a full plan yet on how to replace it but we are trying to remove existing usage and avoid adding new ones. Reflection with config string is an ok workaround for now IMO, or have some builder/ctor pass the instance from your wrapper module.
I am open to a zoom call etc. too if you want to further discuss this.

Reply:
>Are you saying you don't take a dependency on mantis-control-plane-server, so you don't have access to the MasterConfiguration?

There is no issue with that access that I see because later, that just works out with the DI today.